### PR TITLE
Add tzn-mqtt recipe

### PR DESCRIPTION
**Pull request by leonheldattoradex**
tzn-mqtt is a MQTT to DBus bridge. Messages are received by the embedded MQTT client through a mqtt.torizon.io broker, which are then relayed to client applications via DBus.

Tested with qemuarm64.

### CODE REVIEW

**Comment by EdTheBearded** [comment]
Also, you hardcoded that we're currently using dev branch... could we use `use-head-next` here for hashes/branches, as we de in RAC or Aktualizr?

I think that in the future it might come in hand, so new commits are tested ASAP on our nightlies, and then we perform a hash bump on our releases.

**Comment by leonheldattoradex** [comment]
I forgot about `use-head-next` :-)

For the hashes yea, I could do something like this

```
diff --git a/recipes-sota/tzn-mqtt/tzn-mqtt_git.bb b/recipes-sota/tzn-mqtt/tzn-mqtt_git.bb
index f907b6f..9c5abce 100644
--- a/recipes-sota/tzn-mqtt/tzn-mqtt_git.bb
+++ b/recipes-sota/tzn-mqtt/tzn-mqtt_git.bb
@@ -9,7 +9,7 @@ SRC_URI += " \
     file://tzn-mqtt.service \
 "
 
-SRCREV = "516ce5198a077fffcc2a19d946b3c21814841f01"
+SRCREV:use-head-next = "${AUTOREV}"
 S = "${WORKDIR}/git"
 PV = "0.0+git${SRCPV}"
 CARGO_SRC_DIR = ""
```

But for the branches (if I understood correctly): I wouldn't recommend using the main branch. From the development side of the application itself I think it's best if we wait for the e2e test before merging in to `main` and switching, as `main` is not really usable right now on our images.

**Comment by EdTheBearded** [comment]
> I forgot about `use-head-next` :-)
> 
> For the hashes yea, I could do something like this
> 
> ```
> diff --git a/recipes-sota/tzn-mqtt/tzn-mqtt_git.bb b/recipes-sota/tzn-mqtt/tzn-mqtt_git.bb
> index f907b6f..9c5abce 100644
> --- a/recipes-sota/tzn-mqtt/tzn-mqtt_git.bb
> +++ b/recipes-sota/tzn-mqtt/tzn-mqtt_git.bb
> @@ -9,7 +9,7 @@ SRC_URI += " \
>      file://tzn-mqtt.service \
>  "
>  
>  SRCREV = "516ce5198a077fffcc2a19d946b3c21814841f01"
> +SRCREV:use-head-next = "${AUTOREV}"
>  S = "${WORKDIR}/git"
>  PV = "0.0+git${SRCPV}"
>  CARGO_SRC_DIR = ""
> ```
> 
> But for the branches (if I understood correctly): I wouldn't recommend using the main branch. From the development side of the application itself I think it's best if we wait for the e2e test before merging in to `main` and switching, as `main` is not really usable right now on our images.

You guys plan to use the git workflow? With a ongoing dev branch that gets merged from time to time to main, and main being the only branch meant to be used on releases?

**Comment by leonheldattoradex** [comment]
> You guys plan to use the git workflow? With a ongoing dev branch that gets merged from time to time to main, and main being the only branch meant to be used on releases?

My suggestion is to always merge into `main` from the different feature branches instead of `dev` because the main test (beside the unit ones) will be done by Lava, as the repo isn't really a thing on its own.

If you want I can merge to `main` right now and we don't have to switch later from the current `dev` one.

**Comment by EdTheBearded** [comment]
> > You guys plan to use the git workflow? With a ongoing dev branch that gets merged from time to time to main, and main being the only branch meant to be used on releases?
> 
> My suggestion is to always merge into `main` from the different feature branches instead of `dev` because the main test (beside the unit ones) will be done by Lava, as the repo isn't really a thing on its own.
> 
> If you want I can merge to `main` right now and we don't have to switch later from the current `dev` one.

If that's easy enough to do, yeah!

**Comment by leonheldattoradex** [comment]
Let me know how it looks now. I re-tested the changes as well.

File: recipes-images/images/torizon-base.inc [hunk 1]
```diff
@@ -72,6 +72,7 @@ CORE_IMAGE_BASE_INSTALL:append:tdx = " \
 CORE_IMAGE_BASE_INSTALL:append:sota = " \
     auto-provisioning \
     rac \
+    tzn-mqtt \
```
[the remark this answers is not in the document]
Comment (EdTheBearded) [reply recipes-images/images/torizon-base.inc:75]: Checking how Aktualizr is installed, I saw that it is also under the `sota` override. So it is ok that we keep it like this as well!

https://github.com/uptane/meta-updater/blob/master/classes/sota.bbclass#L8

**Review by EdTheBearded — APPROVED** [review]

**Pull request merged**